### PR TITLE
docs: describe npm depTypes

### LINKED ISSUE #4529: properly document npm (engines, volta, etc)
**What would you like Renovate to be able to do?**
Currently different dep types under npm (manager) are not quite well documented like engines and volta.

**Describe the solution you'd like**
Add clear and detailed doc for the same.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nihalwasheretoo**
Added list of various dep types supported by npm manager under `npm` section of docs.
Added `volta` description under node.
Wrapped doc links with description.

Closes #4529

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -594,13 +594,40 @@ Check out our [Node.js documentation](https://docs.renovatebot.com/node) for a c
 
 ## npm
 
+The following dep types are currently supported by the npm manager :
+
+### dependencies
```
Comment (rarkins) [inline docs/usage/configuration-options.md]: No need to list these as subheadings. Just use a regular paragraph or list instead.

**Comment by nihal9ns** [comment]
@rarkins  does it make sense to remove https://github.com/renovatebot/renovate/blob/master/docs/usage/configuration-options.md#engines completely now? as we have it listed under `npm`

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -596,19 +596,12 @@ Check out our [Node.js documentation](https://docs.renovatebot.com/node) for a c
...
-
-### volta
-
-Renovate will update any node and yarn version specified under `volta`.
+1. `dependencies`
+2. `devDependencies`
+3. `optionalDependencies`
+4. `peerDependencies`
+5. `engines`
+6. `volta` : Renovate will update any node and yarn version specified under `volta`.
```
Comment (rarkins) [inline docs/usage/configuration-options.md]: Make this an unordered list. And for engines you can list what we cover there too (node, npm and yarn IIRC)

**Comment by nihal9ns** [comment]
> @rarkins does it make sense to remove https://github.com/renovatebot/renovate/blob/master/docs/usage/configuration-options.md#engines completely now? as we have it listed under `npm`

??

**Comment by rarkins** [comment]
@nihal9ns No, leave it in because it's in definitions and to remove it would be a breaking change.

**Pull request merged**